prov-api/routes/model_run/model_run.py

```diff
@@ -286,46 +287,40 @@ async def update_model_run(
 
     return PostUpdateModelRunResponse(session_id=res)
 
-#
-#
-# class DeleteGraph(BaseModel):
-#    record_id: str
-#
-#
-# @router.post("/delete_graph", operation_id="delete_graph", include_in_schema=False)
-# async def delete_graph(
-#    delete: DeleteGraph,
-#    # admin only for this endpoint
-#    roles: ProtectedRole = Depends(admin_user_protected_role_dependency),
-#    config: Config = Depends(get_settings)
-# ) -> Any:
-#    # no proxy - user direct
-#    request_style = RequestStyle(
-#        service_account=None, user_direct=roles.user)
-#
-#    # dummy delete graph
-#    delete_dummy_graph = NodeGraph(record_id=delete.record_id, links=[])
-#
-#    print("Setting up neo4j client")
-#    neo4j_manager = Neo4jGraphManager(config=config)
-#    print("Done")
-#
-#    # Getting old graph
-#    print("Retrieving graph")
-#    old_graph = neo4j_manager.get_graph_by_record_id(delete.record_id)
-#    print("Done")
-#
-#    # get the graph diff
-#    print("Building diff and applying...")
-#    diff_generator = GraphDiffApplier(neo4j_manager=neo4j_manager)
-#    diff_generator.apply_diff(
-#        old_graph=old_graph, new_graph=delete_dummy_graph)
-#    print("Done")
-#
-#    # get the updated graph
-#    print("Retrieving updated graph")
-#    updated_graph = neo4j_manager.get_graph_by_record_id(delete.record_id)
-#    print("Done")
-#
-#    return {"old_graph": old_graph.to_json_pretty(), "updated": updated_graph.to_json_pretty()}
-#
+
+@router.post("/delete", operation_id="delete_model_run", include_in_schema=False)
+async def delete_model_run(
+    delete: PostDeleteGraphRequest,
+    # admin only for this endpoint
+    roles: ProtectedRole = Depends(admin_user_protected_role_dependency),
+    config: Config = Depends(get_settings)
+) -> None:
+    # grab and ensure it exists and is a model run
+    try:
+        await fetch_item_from_registry_with_subtype(proxy_username=roles.user.username, id=delete.record_id, item_subtype=ItemSubType.MODEL_RUN, config=config)
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Exception: could not delete model run as error occurred when fetching the record from the registry: {e}"
+        ) from e
+
+    # dummy delete graph which is just an empty graph with the record id to delete
+    delete_dummy_graph = NodeGraph(record_id=delete.record_id, links=[])
+
+    print("Setting up neo4j client")
+    neo4j_manager = Neo4jGraphManager(config=config)
+    print("Done")
+
+    # Getting old graph
+    print("Retrieving existing graph")
+    old_graph = neo4j_manager.get_graph_by_record_id(delete.record_id)
+    print("Done")
+
+    # get the graph diff
+    print("Building diff (empty) and applying...")
+    diff_generator = GraphDiffApplier(neo4j_manager=neo4j_manager)
+    diff_generator.apply_diff(
+        old_graph=old_graph, new_graph=delete_dummy_graph)
+    print("Done")
```

Comment:
Cool - just noting i would prefer if there was a trial run mode, where the generated delete/diff actions are returned by the API. But i note that the GraphDiffApplier currently does not expose those, or return anything (assuming no error). and also acknowledge current resource constraints in the team.  So its ok.

Speaking of errors, can you wrap this in a try catch?